### ISSUE TITLE
Notebook "prompt-templates-variables-jinja2" : still uses OpenAI v0 legacy API

### ISSUE THREAD

**Issue by knbknb** [NirDiamant/Prompt_Engineering#3]
I like the elegance of your Notebooks, using Jinja Templates and Langchain from the very beginning.

However, I noticed that the `get_get_completion()` function in `all_prompt_engineering_techniques/prompt-templates-variables-jinja2.ipynb` still uses the OpenAI v0 legacy API. The `requirements.txt` file installs/demands `openai==1.51.1`.

So when  I run the the `class PromptTemplate: ...`  code cell in that Jupyter Notebook I get an error.

```text
APIRemovedInV1:  
You tried to access openai.ChatCompletion, but this is no longer supported in openai>=1.0.0 - see the README at https://github.com/openai/openai-python for the API.  
You can run `openai migrate` to automatically upgrade your codebase to use the 1.0.0 interface.  
Alternatively, you can pin your installation to the old version, e.g. `pip install openai==0.28`
```

I replaced the current code cell defining  `get_get_completion()` function (not reproduced here) with the following code.  So I know how to get along. However I don't know how many Notebooks in this repo will be affected, And I don't know if you prefer to fix everything yourself with the `openai migrate'  mechanism.  That's wh< I'm not sending a pull request.

```python
### my replacement of the  `get_get_completion()` function - new openAI v1 interface
import os
from openai import OpenAI 
from jinja2 import Template
from dotenv import load_dotenv

load_dotenv()

def get_completion(prompt, model="gpt-4o-mini"):
    ''' Get a completion from the OpenAI API 
    Args:
        prompt (str): The prompt to send to the API
        model (str): The model to use for the completion
    Returns:
        str: The completion text
    '''
    messages = [{"role": "user", "content": prompt}]
    client = OpenAI(api_key=os.getenv('OPENAI_API_KEY'))
    response = client.chat.completions.create(
        model=model,
        messages=messages,
        temperature=0,
    )
    return response.choices[0].message.content
```

**Comment by NirDiamant** [comment]
Thanks for the heads up!
What I suggest is to add a line at the beginning of this tutorial pip
installing the relevant packages only for this one, and then applying your
changes.
I would love to accept your PR

**Comment by NirDiamant** [comment]
You're right - several notebooks still use the legacy OpenAI v0 API (openai.ChatCompletion.create instead of the new client pattern). This is a known issue. Updating them to the v1+ API is on the list. PRs welcome if anyone wants to help migrate specific notebooks.

**Comment by NirDiamant** [comment]
Thanks for the careful report, and apologies it sat this long. Good news: this is fixed.

I checked `all_prompt_engineering_techniques/prompt-templates-variables-jinja2.ipynb` on `main` and it now uses the v1 client API throughout:

```python
from openai import OpenAI
client = OpenAI()

def get_completion(prompt, model="gpt-4o-mini"):
    messages = [{"role": "user", "content": prompt}]
    response = client.chat.completions.create(model=model, messages=messages, temperature=0)
    return response.choices[0].message.content
```

No `openai.ChatCompletion` anywhere, so the `APIRemovedInV1` error you hit is gone. I also grepped the rest of the repo for the legacy call and found no remaining instances, which answers your "how many notebooks are affected" question: none.

To your point about whether you should have sent a PR - you should feel free to, next time. A fix plus a note on scope is exactly the right shape for a PR here, and I would rather review that than have you hold back. Thanks for the kind words about the notebooks.
